js/coinone.js

```diff
@@ -0,0 +1,298 @@
+            'currency': mkt['id'],
+            'format': 'json',
+        }, params));
+        let ticker = res;
+        let baseVolume = this.safeFloat (ticker, 'volume');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'last'),
+            'ask': this.safeFloat (ticker, 'last'),
+            'vwap': undefined,
+            'open': undefined,
+            'close': undefined,
+            'first': this.safeFloat (ticker, 'first'),
+            'last': this.safeFloat (ticker, 'last'),
+            'change': this.safeFloat (ticker, 'yesterday_last') - this.safeFloat (ticker, 'last'),
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': baseVolume,
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    parseTrade (trade, symbol = undefined) {
+        let timestamp = parseInt (trade['timestamp']) * 1000;
+        let side = 'buy';
+        return {
+            'id': undefined,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'order': undefined,
+            'symbol': symbol,
+            'type': undefined,
+            'side': side,
+            'price': this.safeFloat (trade, 'price'),
+            'amount': this.safeFloat (trade, 'qty'),
+            'fee': undefined,
+            'info': trade,
+        };
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let market = this.market (symbol);
+        let result = await this.publicGetTrades (this.extend ({
+            'currency': market['id'],
+            'period': 'hour',
+            'format': 'json',
+        }, params));
+        let trades = result['completeOrders'];
+        return this.parseTrades (trades, symbol);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        // throw new ExchangeError (this.id + ' cancelOrder () is not fully implemented yet');
+        let method = 'privatePostOrderCancel';
+        return await this[method] ({ 'orderID': id });
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let request = this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        let url = this.urls['api'] + '/' + request;
+        headers = {};
+        if (api === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            let nonce = this.nonce ().toString ();
+            let payload = this.stringToBase64 (JSON.stringify ({ 'access_token': this.apiKey, 'nonce': nonce }));
```

Comment:
Also, we might have a problem with `JSON.stringify ({ ... })` here, because of it we might need to switch to use base method `this.json ({ ... })` here instead.